test/unit/timertask-spec.ts

```diff
@@ -1,30 +1,44 @@
 import { expect } from "chai";
 import "mocha";
 import { CreateTimerAction } from "../../src/classes";
-import { TaskFactory } from "../../src/tasks/taskfactory";
+import { TimerTask } from "../../src/tasks/externalTasks";
+import { InnerTimerTask } from "../../src/tasks/internalTasks";
 
 describe("TimerTask", () => {
     it("throws cannot cancel a completed task", async () => {
-        const task = TaskFactory.CompletedTimerTask(
-            new CreateTimerAction(new Date(), false),
-            new Date(),
-            0,
-            5
-        );
+        const isCancelled = false;
+        const date = new Date();
+        const action = new CreateTimerAction(date, isCancelled);
+        const innerTask = new InnerTimerTask(0, action);
+        innerTask.SetValue(false, undefined); // set value to complete task
+
+        const task = new TimerTask(innerTask);
+        task.internalTask.SetValue(false, undefined);
+
         expect(() => {
             task.cancel();
         }).to.throw("Cannot cancel a completed task.");
     });
 
     it("cancels an incomplete task", async () => {
-        const task = TaskFactory.UncompletedTimerTask(new CreateTimerAction(new Date()));
+        const isCancelled = false;
+        const date = new Date();
+        const action = new CreateTimerAction(date, isCancelled);
+        const innerTask = new InnerTimerTask(0, action);
+
+        const task = new TimerTask(innerTask);
         task.cancel();
-        expect(task.action.isCanceled).to.equal(true);
-        expect(task.isCanceled).to.equal(true);
+        expect(task.isCancelled).to.equal(true);
+        expect(task.isCancelled).to.equal(true);

```

Comment:
Why are we running the same `expect` call twice?

Reply:
This was the result of some fancy auto-refactoring. I've removed the duplicate assertion.